cl/singlenode/singlenode.go

```diff
@@ -329,24 +339,22 @@ func (app *SingleNodeApp) produceBlock() error {
 	}
 
 	if app.payloadRepo != nil {
-		payloadInfo := &types.PayloadInfo{
+		// Save payload to repository
+		saveCtx, saveCancel := context.WithTimeout(app.appCtx, 400*time.Millisecond)
```

Comment:
Do we intend to double the timeout here?

Reply:
It usually finishes early. Seems safer. I can change it back if you think its not needed.